cpp/src/interop/from_arrow_device.cu

```diff
@@ -317,6 +320,11 @@ dispatch_tuple_t get_column(ArrowSchemaView* schema,
                             rmm::cuda_stream_view stream,
                             rmm::device_async_resource_ref mr)
 {
+  CUDF_EXPECTS(
+    input->length <= static_cast<std::size_t>(std::numeric_limits<cudf::size_type>::max()),
```

Comment:
```suggestion
    input->length <= static_cast<int64_t>(std::numeric_limits<cudf::size_type>::max()),
```

Reply:
Is this because `input->length` is an `int64_t`? Should one then rather not cast to `decltype(input->length)`? FWIW, I think the usual arithmetic promotions would mean we don't need a static_cast here at all.

Reply:
Yes, I think  you can remove the `static_cast` since we are comparing signed/signed here.

Reply:
https://godbolt.org/z/66cbbd4d9

Reply:
Thanks, makes sense. I will leave the explicit casts in (since I always have to look up integer promotion rules).